File: modified_scripts/scala/matrix_multiplication/qwen3_8b.scala
def matrixMultiplication(a: Array[Array[Int]], b: Array[Array[Int]]): Array[Array[Int]] = {
  val rowsA = a.length
  val colsA = a(0).length
  val rowsB = b.length
  val colsB = b(0).length
  val result = Array.ofDim[Int](rowsA, colsB)
  for (i <- 0 until rowsA) {
    for (j <- 0 until colsB) {
      var sum = 0
      for (k <- 0 until colsA) {
        sum += a(i)(k) * b(k)(j)
      }
      result(i)(j) = sum
    }
  }
  result
}

File: original_scripts/r/matrix_multiplication/qwen3_8b.r
matrix_multiplication <- function(a, b) {
  n <- nrow(a)
  m <- ncol(a)
  p <- ncol(b)
  result <- matrix(0, n, p)
  for (i in 1:n) {
    for (j in 1:p) {
      for (k in 1:m) {
        result[i, j] <- result[i, j] + a[i, k] * b[k, j]
      }
    }
  }
  result
}